### PR TITLE
Add OS to template create request

### PR DESCRIPTION
**Pull request by mattkelly**
#### What does this pull request accomplish?

Add OS to template create request

 #### What issue(s) does this fix?
 None

 #### Additional Considerations

 ### Testing

```
csctl create template do --description "testtttt" -m3 -w2 --os centos --debug
ID                                      DESCRIPTION     PROVIDER_NAME   MASTER_VERSION  OWNER_ID                                CREATED_AT
e106fb2c-8309-479d-bc3a-52c468aa5192    testtttt        digital_ocean   1.12.3          576eb5eb-26df-47af-8abc-f4effcd59ec1    2019-04-18 13:14:05 -0400 EDT
```

 ### Dependencies
 None

### CODE REVIEW

File: cloud/provision/swagger.yaml [hunk 1]
```diff
@@ -1617,6 +1615,10 @@ definitions:
         type: integer
         format: int32
         minimum: 1
+      os:
+        description: Operating System
+        type: string
+        enum: [centos, ubuntu]
```
Comment (normanjoyner) [inline cloud/provision/swagger.yaml]: are we holding off on RHEL since DigitalOcean does not support?

Comment (mattkelly) [reply cloud/provision/swagger.yaml]: No harm in adding it now, just didn't really think about it. We don't actually use these for validation anywhere within csctl yet anyway :/

File: resource/options/template_create_digitalocean.go [hunk 2]
```diff
@@ -85,8 +85,14 @@ func (o *DigitalOceanTemplateCreate) digitalOceanDropletConfiguration() types.Di
 }
 
 func (o *DigitalOceanTemplateCreate) defaultAndValidateImage() error {
+	// Assumes o.TemplateCreate.DefaultAndValidate() already called
```
Comment (mattkelly) [inline resource/options/template_create_digitalocean.go]: The more stuff like this leaks in, the more I dislike this pattern. Food for thought for the future.

File: cloud/provision/swagger.yaml (hunk 1 above)
Comment (mattkelly) [reply cloud/provision/swagger.yaml]: Updated

**Comment by mattkelly** [comment]
As discussed in Slack, I added another commit to allow code coverage to drop up to 1% for the project. If it works, this should be good to go.

**Pull request merged**